src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.db.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        Storage.storage.merge(fruitTransaction.getFruit(),
+                -fruitTransaction.getQuantity(), Integer::sum);
```

Also ensure exceptions and invalid situations are surfaced with informative RuntimeExceptions rather than allowing NPEs or silent failures. The checklist requires: "Don't ignore exceptions in try-catch blocks ... Let's rethrow a RuntimeException with an informative message and exception object." Wrap any checked parsing or validation-related exceptions at call sites and throw meaningful RuntimeExceptions that include context (fruit name, attempted quantity, etc.).